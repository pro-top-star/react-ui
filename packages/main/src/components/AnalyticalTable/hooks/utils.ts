export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
