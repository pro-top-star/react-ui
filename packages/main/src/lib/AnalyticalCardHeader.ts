import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
