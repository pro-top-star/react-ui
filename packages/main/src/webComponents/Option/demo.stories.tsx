import React from 'react';

import { boolean } from '@storybook/addon-knobs';
import { Option } from './index';

export default {
  title: 'UI5 Web Components | Option',
  component: Option
};

export const generatedDefaultStory = () => (
  <Option selected={boolean('selected', false)} icon={'sap-icon://add'} value={'generatedString'} />
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
