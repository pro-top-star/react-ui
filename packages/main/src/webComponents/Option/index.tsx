import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Option from '@ui5/webcomponents/dist/Option';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface OptionPropTypes extends WithWebComponentPropTypes {
  selected?: boolean; // @generated
  icon?: string; // @generated
  value?: string; // @generated
}

/**
 * <code>import { Option } from '@ui5/webcomponents-react/lib/Option';</code>
 */
const Option: FC<OptionPropTypes> = withWebComponent<OptionPropTypes>(UI5Option);

Option.displayName = 'Option';

Option.defaultProps = {
  icon: null // @generated
};

export { Option };
