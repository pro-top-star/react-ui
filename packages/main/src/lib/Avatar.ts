import { Avatar } from '../webComponents/Avatar';

export { Avatar };
