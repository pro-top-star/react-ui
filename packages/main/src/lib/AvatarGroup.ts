import { AvatarGroup } from '../webComponents/AvatarGroup';
import type { AvatarGroupPropTypes } from '../webComponents/AvatarGroup';

export { AvatarGroup };
export type { AvatarGroupPropTypes };
