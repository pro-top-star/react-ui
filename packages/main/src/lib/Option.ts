import { Option } from '../webComponents/Option';

export { Option };
