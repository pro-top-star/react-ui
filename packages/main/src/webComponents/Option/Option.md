```jsx
import { Option } from '@ui5/webcomponents-react/lib/Option';
```
