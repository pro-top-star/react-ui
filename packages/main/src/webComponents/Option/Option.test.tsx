import { mountThemedComponent } from '@shared/tests/utils';
import { Option } from '@ui5/webcomponents-react/lib/Option';
import React from 'react';

describe('Option', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Option />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
