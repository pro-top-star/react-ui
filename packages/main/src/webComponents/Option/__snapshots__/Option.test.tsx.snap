// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option Basic Test (generated) 1`] = `<ui5-option />`;
